File: src/P006/p006.py
from itertools import combinations


def solve1() -> int:
    n = 100  # pylint: disable=invalid-name
    sum_of_squares = (n * (n + 1) * (2 * n + 1)) // 6
    square_of_sum = ((n * (n + 1)) // 2) ** 2
    return square_of_sum - sum_of_squares


def solve() -> int:
    return 2 * sum(i * j for i, j in combinations(range(1, 101), 2))

File: src/P006/p006.ts
// Brute force implementation
export const solve1 = (): string => {
  const n = 100;
  let sumOfSquares = 0;
  let squareOfSum = 0;
  for (let i = 1; i <= n; i++) {
    sumOfSquares += i * i;
    squareOfSum += i;
  }

  squareOfSum *= squareOfSum;
  return (squareOfSum - sumOfSquares).toString();
};

// Direct computation
export const solve = (): string => {
  const n = 100;
  const sumOfSquares = (n * (n + 1) * (2 * n + 1)) / 6;
  const squareOfSum = ((n * (n + 1)) / 2) ** 2;
  return (squareOfSum - sumOfSquares).toString();
};
